flocker/control/test/test_config.py

```diff
@@ -2666,17 +2666,24 @@ def test_error_on_unrecognized_application_name(self):
 
     def test_set_on_success(self):
         """
-        ``_deployment_from_config`` returns a set of ``Node`` objects. One for
-        each key in the supplied nodes dictionary.
+        ``deployment_from_configuration`` returns a set of ``Node``
+        objects. One for each key in the supplied nodes dictionary.
         """
+        manifestation = Manifestation(
+            dataset=Dataset(
+                dataset_id=dataset_id_from_name(
+                    "mysq-hybridcluster"),
+                metadata=pmap(
+                    {"name": "mysql-hybridcluster"})),
+            primary=True)
+
         applications = {
             'mysql-hybridcluster': Application(
                 name='mysql-hybridcluster',
-                image=Application(
-                    name='mysql-hybridcluster',
-                    image=DockerImage(repository='flocker/mysql',
-                                      tag='v1.0.0'))
-            )
```

Comment:
So was this test broken? The diff makes it look like we used to supply an `Application` as the `image` attribute of an `Application`.